source/common/http/path_utility.cc

```diff
@@ -52,5 +54,20 @@ bool PathUtil::canonicalPath(HeaderEntry& path_header) {
   return true;
 }
 
+void PathUtil::mergeSlashes(HeaderEntry& path_header) {
+  const auto original_path = path_header.value().getStringView();
+  // Only operate on path component in URL.
+  const absl::string_view::size_type query_start = original_path.find('?');
+  const absl::string_view path = original_path.substr(0, query_start);
+  const absl::string_view query = absl::ClippedSubstr(original_path, query_start);
+  if (path.find("//") == absl::string_view::npos) {
+    return;
+  }
+  const absl::string_view prefix = absl::StartsWith(path, "/") ? "/" : absl::string_view();
+  const absl::string_view suffix = absl::EndsWith(path, "/") ? "/" : absl::string_view();
+  path_header.value(absl::StrCat(
```

Comment:
I assume this is O(n) if we get something whacky like a path which is 16k worth of / yeah?

Reply:
I don't see anything explicit in the absl documentation, but it's O(n) based on the code.